docs/RFCs/package-toml.md

```diff
@@ -0,0 +1,128 @@
+# Package.toml format
+
+A proposal for a more formal Package.toml definition
+
+- Version: DRAFT
+- Date: 06/06/2019
+- Authors
+  - Todd Kennedy <todd@selfassembled.org>
+
+## Abstract
+
+This document is meant to describe the package manifest format for Entropic,
+a federated solution to package management. Entropic needs to have its own
+manifest file due to conflicts arising from the namespacing entropic uses.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Copyright
+
+This document is copyright by its individual contributors, and available for
+re-use under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)
+
+## Introduction
+
+Entropic cannot share a manifest format with the traditional `package.json` format
+associated with the ECMAScript ecosystem for a number of reasons:
+
+1. Entropic is intended to allow for federated package management for more than just the ECMAScript ecosystem
+1. The `package.json` name specifier does not allow for the namespacing that entropic provides
+1. There are recognized deficiencies with using JSON as human-readable metadata format
+
+## Filename
+
+The filename of the manifest **MUST** be `Package.toml` and it **MUST** live in
+the root directory of the package's filesystem. The root of the package is defined
+as the closest parent directory that contains the manifest file, starting with
+the current directory.
+
+On case-insensitive filesystems, it **MAY** be presented as `package.toml` but care
+should be taken to preserve case when possible.
+
+## Format
+
```

Comment:
Curious about the decision to use TOML instead of JSON. I’ve seen tremendous value in the ability to parse package.json files “natively” in node. Maybe worth describing the motivation for this decision as I am sure it will be a common question.

Reply:
Comments, I expect. I've sometimes wanted to add a comment explaining why a dependency is pinned at a specific version, or add a description to a "scripts" command, but JSON doesn't allow that.

You could go with something like [JSON5](https://json5.org) but then you lose the ability to parse it "natively" in Node, and at that point it doesn't really matter what you use.

Reply:
There was some discussion about why TOML in https://github.com/entropic-dev/entropic/issues/106 and https://github.com/entropic-dev/entropic/issues/162, in particular https://github.com/entropic-dev/entropic/issues/106#issuecomment-497982913 and https://github.com/entropic-dev/entropic/issues/162#issuecomment-499264287

Reply:
Those issues don't really explain why TOML was chosen, only why Entropic doesn't use package.json. Although I support the use of TOML, there's no reason the package file couldn't be entropic.json instead.

Reply:
JSON doesn't support comments which has been a frequent complaint about it (especially when used for manifests). I'm also adding a "why TOML" section to the RFC